app/controllers/charges_controller.rb

```diff
@@ -1,25 +1,37 @@
 class ChargesController < ApplicationController
   def new
+    order = Order.find_by(id: session[:checkout_order_id])
+    if order
+      @amount = order.total
+    else
+      redirect_to books_path, alert: "You must buy something to check out"
+    end
   end
 
   def create
-    # Amount in cents
-    @amount = cart.stripe_total
+    order = Order.find_by(id: session[:checkout_order_id])
+    if order
+      # Amount in cents
+      @amount = order.stripe_total
```

Comment:
Does this need to be an instance variable? It looks like it's only used within this method.